packages/sentry/src/adapters/sentryMonitoringAdapter.ts

```diff
@@ -0,0 +1,161 @@
+   *
+   * @default 0.025
+   */
+  tracesSampleRate: number;
+  /**
+   * Custom tags to add in all transaction.
+   *
+   */
+  customTags?: { [tag: string]: string };
+}
+
+const defaultOptions: SentryMonitoringConfigs = {
+  dsn: '',
+  environment: 'local',
+  debug: false,
+  tracesSampleRate: 0.025,
+  release: '',
+};
+
+/**
+ * This is a Sentry Monitoring Adapter that logs element in the Sentry.
+ * There are four methods implemented which are `setUserId`, `log`, `handleError` and `createTransaction`,
+ *
+ * At the initialization level, user have to pass required sentry configuration keys to make connection between
+ * application and Sentry. `log` and `handleError` methods will log data and errors respectively in the Sentry dashboards.
+ * While `createTransaction` method used to measure a performance of an application at any given point.
+ */
+export class SentryMonitoringAdapter implements MonitoringAdapter {
+  constructor(optionsParam: Partial<SentryMonitoringConfigs>) {
+    Sentry.init({
+      dsn: optionsParam.dsn || '',
+      environment: optionsParam.environment ?? 'local',
+      debug: optionsParam.debug ?? true,
+      release: optionsParam.release ?? '1.0',
+      tracesSampleRate: optionsParam.tracesSampleRate ?? 0.025,
+      beforeBreadcrumb(breadcrumb, hint) {
+        if (breadcrumb.category === 'xhr') {
+          return null;
+        }
+
+        return breadcrumb;
+      },
+    });
+
+    if (optionsParam.customTags) {
+      Sentry.setTags(optionsParam.customTags);
+    }
+  }
+
+  setUserId(id: string): void {
+    Sentry.setUser({ id });
+  }
+
+  log(msg: string, context?: LogContext | Severity): void {
+    Sentry.captureMessage(msg, context);
+  }
+
+  handleError(err: Error | string, context?: Omit<LogContext, 'level'>): void {
+    Sentry.captureException(err, context);
+  }
+
+  createTransaction(context: TransactionContext): Transaction {
+    // Create a new sentry transaction
+    const transaction = Sentry.startTransaction({
+      name: context.name || '',
+      data: context.data || {},
+      op: context.operation || '',
+      description: context.description || '',
+      traceId: context.traceId || '',
+      tags: context.tags || {},
+      sampled: true,
+    });
+
+    // Create an object to map spans for a transaction
+    const transactionSpansObj: { [key: string]: any } = {};
+
+    return {
+      setTag: (tagName: string, tagValue: string) => transaction.setTag(tagName, tagValue),
+      startMeasurement: (name: string, data?: { [key: string]: number | string }) => {
+        transactionSpansObj[name] = transaction.startChild({
+          op: name,
+          data: data || {},
+        });
+      },
+      stopMeasurement: () => (name: string) => {
+        if (transactionSpansObj[name]) {
+          const spanObj: Span = transactionSpansObj[name] as Span;
+          spanObj.setStatus(MonitoringStatus.OK);
+          spanObj.finish();
+          delete transactionSpansObj[name];
```

Comment:
Since you should have defined the type `Span` in the `transactionSpans` record based on my previous comment, you can change these lines to just :

```typescript
transactionSpans[name].setStatus(MonitoringStatus.OK);
transactionSpans[name].finish();
delete transactionSpans[name];
```